test/kitchen/tasks/kitchen.py

```diff
@@ -0,0 +1,156 @@
+from invoke import task
+from invoke.exceptions import Exit
+
+import glob
+import json
+import os.path
+import re
+
+@task(iterable=['platlist'])
+def genconfig(
+    ctx,
+    platform=None,
+    provider=None,
+    osversions="all",
+    testfiles=None,
+    uservars=None,
+    platformfile="platforms.json",
+    platlist=None
+):
+    """
+    Create a kitchen config
+    """
+    if not platform and not platlist:
+        print("Must supply a platform to configure\n")
+        raise Exit(1)
+
+    if not testfiles:
+        print("Must supply one or more testfiles to include\n")
+        raise Exit(1)
+
+    if platlist and (platform or provider):
+        print("Can specify either a list of specific OS images OR a platform and provider, but not both\n")
+        raise Exit(1)
+
+    if not platlist and not provider:
+        provider = "azure"
+
+    platforms = load_platforms(ctx, platformfile=platformfile)
+
+    # create the TEST_PLATFORMS environment variable
+    testplatforms = ""
```

Comment:
I think this is unnecessary since `testplatforms` will be declared inside the `if platform` or the `elif platlist` branches ([Python does not scope variables to control blocks](https://stackoverflow.com/a/2829642) like Go does)
```suggestion
```

Reply:
There is as path where it's used before creating it w/o the forward declaration.
The `if testplatforms` in both the if and elif block occurs before the first assignment.